iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/fifo.inl

```diff
@@ -71,7 +71,7 @@ inline bool FiFo<ValueType, Capacity>::empty() const noexcept
 template <class ValueType, uint64_t Capacity>
 inline cxx::optional<ValueType> FiFo<ValueType, Capacity>::pop() noexcept
 {
-    auto currentReadPos = m_read_pos.load(std::memory_order_relaxed);
+    auto currentReadPos = m_read_pos.load(std::memory_order_acquire);
```

Comment:
Ok, no previous loads can move after this and no later reads/writes in this function before it. I think the latter is guaranteed either way by dependency. E.g. the next line (basically everything containing `currentReadPos`) cannot move before this (as it depends on the load).

This should is fine and we need the acquire to synchronize with the `m_read_pos.store` of a concurrent `pop` call (i.e. we cannot relax it).

Comment for common understanding. Resolve if this is the intention.